tests/test_types.py

```diff
@@ -145,3 +146,106 @@ def test_pattern_negated_character_range(self) -> None:
         self.assertFalse(selector.matches(self._make_html_view("example.7")))
         self.assertFalse(selector.matches(self._make_html_view("example.8")))
         self.assertFalse(selector.matches(self._make_html_view("example.9")))
+
+    def test_basescope2languageid(self) -> None:
+        self.assertEqual(basescope2languageid("source.js.vite"), "javascript")
+        self.assertEqual(basescope2languageid("source.c++"), "cpp")
+        self.assertEqual(basescope2languageid("source.coffee.gulpfile"), "coffeescript")
+        self.assertEqual(basescope2languageid("source.cs"), "csharp")
+        self.assertEqual(basescope2languageid("source.css.tailwind"), "css")
+        self.assertEqual(basescope2languageid("source.dosbatch"), "bat")
+        self.assertEqual(basescope2languageid("source.fixedform-fortran"), "fortran")
+        self.assertEqual(basescope2languageid("source.groovy.gradle"), "groovy")
+        self.assertEqual(basescope2languageid("source.groovy.jenkins"), "groovy")
+        self.assertEqual(basescope2languageid("source.js"), "javascript")
+        self.assertEqual(basescope2languageid("source.js.eslint"), "javascript")
+        self.assertEqual(basescope2languageid("source.js.gruntfile"), "javascript")
+        self.assertEqual(basescope2languageid("source.js.gulpfile"), "javascript")
+        self.assertEqual(basescope2languageid("source.js.postcss"), "javascript")
+        self.assertEqual(basescope2languageid("source.js.puglint"), "javascript")
+        self.assertEqual(basescope2languageid("source.js.react"), "javascriptreact")
+        self.assertEqual(basescope2languageid("source.js.stylelint"), "javascript")
+        self.assertEqual(basescope2languageid("sourcet.js.unittest"), "javascript")
+        self.assertEqual(basescope2languageid("source.js.webpack"), "javascript")
+        self.assertEqual(basescope2languageid("source.json-tmlanguage"), "jsonc")
+        self.assertEqual(basescope2languageid("source.json.babel"), "json")
+        self.assertEqual(basescope2languageid("source.json.bower"), "json")
+        self.assertEqual(basescope2languageid("source.json.composer"), "json")
+        self.assertEqual(basescope2languageid("source.json.eslint"), "json")
+        self.assertEqual(basescope2languageid("source.json.npm"), "json")
```

Comment:
Having **all** those tests might be missing the point a bit. If we want to test that the function does proper matching then we should only need to test "unique" inputs from the point of the function. And should then likely mock the map so that it's clear from the test point of view what are all the inputs.

Unless we are testing **the mappings** and not the function itself which looks more like what we are doing here. But that doesn't do much for unittesting the code itself and it won't really help to catch the cases we don't account for now.

In any case, we can keep it but we can rewrite those with a lot less duplicated code.

Reply:
I just wrote the test to test the mapping to  guarantee that we don't miss some things.

But a agree that we can remove it, before we merge.